Engine/src/creature.cpp

```diff
@@ -44,6 +43,11 @@ Creature::Creature(neat::Genome genome, Mutable mutables)
     size_ = mutables.GetBabySize();
     health_ = mutables.GetIntegrity() * pow(size_, 2);
     energy_ = mutables.GetEnergyDensity() * pow(size_, 2);
+    int neural_inputs = 0;
+    for (BrainModule module : genome.GetModules()){
+        neural_inputs += module.GetInputNeuronIds().size();
+    }
+    neuron_data_ = std::vector<double>(neural_inputs, 0.0);
```

Comment:
I am not completely familiar with the creature class but it seems like the neural_inputs is equal to the number of input neurons from modules. What about input neurons that are not a part of any module?